can/simlib/transport.cpp

```diff
@@ -0,0 +1,45 @@
+#include "can/simlib/transport.hpp"
+
+#include "common/core/freertos_synchronization.hpp"
+#ifdef USE_SOCKETCAN
+#include "can/simlib/socketcan_transport.hpp"
+#else
+#include "can/simlib/socket_transport.hpp"
+#endif
+
+/**
+ * Create simulating bus transport
+ * @return pointer to bus transport
+ */
+auto can_transport::create()
+    -> std::shared_ptr<can_transport::BusTransportBase> {
+#ifdef USE_SOCKETCAN
```

Comment:
if we're building this in to the same function and using getenv anyway, should we just make this fully runtime-dynamic? or can SocketCanTransport not compile without linux?

Reply:
I wish. We cannot compile `SocketCanTransport` on non linux machine.